dashboards-observability/public/components/visualizations/charts/data_table/data_table.tsx

```diff
@@ -3,102 +3,487 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import React, { useState, useMemo, useCallback } from 'react';
-import { EuiInMemoryTable, EuiDataGrid } from '@elastic/eui';
+import React, { useState, useMemo, useCallback, useRef, useEffect } from 'react';
+import {
+  EuiButtonIcon,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButtonEmpty,
+  EuiPopover,
+  EuiSwitch,
+  EuiIcon,
+  EuiContextMenuPanel,
+  EuiContextMenuItem,
+} from '@elastic/eui';
 
-export const DataTable = ({ visualizations }: any) => {
+// pagination
+import ReactPaginate from 'react-paginate';
+
+// af-data-grid
+import { AgGridReact } from 'ag-grid-react';
+import 'ag-grid-community/dist/styles/ag-grid.css';
+import 'ag-grid-community/dist/styles/ag-theme-alpine.css';
+import 'ag-grid-community/dist/styles/ag-theme-alpine-dark.css';
+
+// styles
+import './data_table.scss';
+
+// theme
+import { uiSettingsService } from '../../../../../common/utils';
+
+export const DataTable = ({ visualizations, layout, config }: any) => {
   const {
     data: vizData,
     jsonData,
     metadata: { fields = [] },
   } = visualizations.data.rawVizData;
-
   const raw_data = [...jsonData];
+  const columns = fields.map((field: any) => {
+    return {
+      headerName: field.name,
+      field: field.name,
+      id: field.name,
+      colId: field.name,
+      lockVisible: true,
+      columnsMenuParams: {
+        suppressColumnFilter: true,
+        suppressColumnSelectAll: true,
+        suppressColumnExpandAll: true,
+      },
+    };
+  });
 
-  // Pagination
-  const [pagination, setPagination] = useState({ pageIndex: 0, pageSize: 10 });
-  const onChangeItemsPerPage = useCallback(
-    (pageSize) =>
-      setPagination((pagination) => ({
-        ...pagination,
-        pageSize,
-        pageIndex: 0,
-      })),
-    [setPagination]
-  );
-  const onChangePage = useCallback(
-    (pageIndex) => setPagination((pagination) => ({ ...pagination, pageIndex })),
-    [setPagination]
-  );
+  // ag-grid-react bindings
+  const gridRef = useRef<any | undefined>();
+  const gridRefFullScreen = useRef<any | undefined>();
+  const [pageSize, setPageSize] = useState(10);
+  const [columnVisibility, setColumnVisibility] = useState([]);
+  const [isFullScreen, setIsFullScreen] = useState<boolean>(false);
+  const [selectedRowDensity, setSelectedRowDensity] = useState({
+    icon: 'tableDensityNormal',
+    height: 40,
+    selected: true,
+  });
+  const defaultColDef = useMemo(() => {
+    return {
+      editable: true,
+      sortable: true,
+      resizable: true,
+      filter: true,
+      flex: 1,
+      minWidth: 100,
+      suppressMenu: true,
+    };
+  }, []);
 
-  // Sorting
-  const [sortingColumns, setSortingColumns] = useState([]);
-  const onSort = useCallback(
-    (sortingColumns) => {
-      setSortingColumns(sortingColumns);
+  const onPageSizeChanged = useCallback(
+    (val) => {
```

Comment:
This can be assigned a type. Looks like a number.

Reply:
okay